include/sparrow/memory.hpp

```diff
@@ -0,0 +1,94 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or mplied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cassert>
+#include <memory>
+
+namespace sparrow {
+
+/**
+ * @brief A value_ptr is a smart pointer that behaves like a value.
+ * It manages the lifetime of an object of type T.
```

Comment:
```suggestion
 * It manages the lifetime of an object of type T which is not stored in the `value_ptr` but a pointer, similar to `unique_ptr`.
```

or some clarification in that direction.